src/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Sinusoid.hpp

```diff
@@ -49,8 +53,14 @@ class Sinusoid : public MarkAsAnalyticSolution {
       const tnsr::I<DataType, 1>& x, double t,
       tmpl::list<ScalarAdvection::Tags::U> /*meta*/) const;
 
-  // NOLINTNEXTLINE(google-runtime-references)
-  void pup(PUP::er& p);
+  // clang-tidy: no pass by reference
+  void pup(PUP::er& p);  // NOLINT
+
+  /// \cond
+  explicit Sinusoid(CkMigrateMessage* msg) : InitialData(msg) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(Sinusoid);
+  /// \endcond
 };
```

Comment:
Can you add a test for each, something like
```cpp
const std::unique_ptr<InitialData> id = std::make_unique<Sinusoid>();
const std::unique_ptr<InitialData> deserialized_id = serialize_and_deserialize(id);
CHECK(dynamic_cast<Sinusoid*>(deserialized_id.get()) != nullptr);
```
but maybe with better variable names :) This is exactly the thing you want to have working, so I think it's important to test that.